### ISSUE TITLE
Maliar algorithm to intelligently iterate over sampling and training

### ISSUE THREAD

**Issue by sbenthall** [scikit-agent/scikit-agent#47]
With #41 and #46 (but also with the existing lifetime reward estimation loss function) , there should be an algorithm that iterates between sampling and training to find decision rule parameters that bring the loss to a sufficiently tolerable level, as per MMW JME '21.
